components/layout/display_list_builder.rs

```diff
@@ -673,6 +697,35 @@ fn convert_gradient_stops(gradient_items: &[GradientItem],
     stops
 }
 
+#[inline]
+/// Duplicate the first and last stops if necessary.
+///
+/// Explanation by pyfisch:
+/// If the last stop is at the same position as the previous stop the
+/// last color is ignored by webrender. This differs from the spec
+/// (I think so). The  implementations of Chrome and Firefox seem
+/// to have the same problem but work fine if the position of the last
+/// stop is smaller than 100%. (Otherwise they ignore the last stop.)
+///
+/// Similarly the first stop is duplicated if it is not placed
+/// at the start of the virtual gradient ray.
```

Comment:
> If multiple color stops have the same position, they produce an infinitesimal transition from the one specified first in the rule to the one specified last. In effect, the color suddenly changes at that position rather than smoothly transitioning.

Well this is what the spec says about color stops with the same position. There are no special rules for the first and last stop. The way I understand the spec testcase 12 should have a green background with the circle in the middle like example 11 but bigger. Unfortunately  this is not what is implemented in Blink and Gecko. Maybe there is a note in the spec but I missed it completely. :confused: 

servo/webrender#1189 seems to do something similiar but on the webrender side. (I have not tested if it actually works like this function.